### PR TITLE
Add positive chill units

### PR DESCRIPTION
**Pull request by saschahofmann**
### Pull Request Checklist:

- [x] Tests for the changes have been added (for bug fixes / features)
- [x] CHANGELOG.rst has been updated (with summary of main changes)

### What kind of change does this PR introduce?

* Adds a new argument to `chill_unit` `positive_only` to compute the daily positive chill units

### CODE REVIEW

File: xclim/indices/_agro.py [hunk 1]
```diff
@@ -1680,4 +1684,7 @@ def chill_units(tas: xarray.DataArray, freq: str = "YS") -> xarray.DataArray:
         ),
     )
     cu = cu.where(tas.notnull())
-    return cu.resample(time=freq).sum().assign_attrs(units="")
+    daily = cu.resample(time="1D").sum()
+    if positive_only:
+        daily = daily.where(daily > 0)
+    return daily.resample(time=freq).sum().assign_attrs(units="")
```
Comment (aulemahal) [inline xclim/indices/_agro.py:1690]:
```suggestion
    if positive_only:
        daily = cu.resample(time="1D").sum()
        cu = daily.where(daily > 0)
    return cu.resample(time=freq).sum().assign_attrs(units="")
```
This way we can avoid adding operations when not needed?

File: xclim/indices/_agro.py [hunk 2]
```diff
@@ -1635,17 +1635,21 @@ def chill_portions(
 
 
 @declare_units(tas="[temperature]")
-def chill_units(tas: xarray.DataArray, freq: str = "YS") -> xarray.DataArray:
+def chill_units(
+    tas: xarray.DataArray, positive_only=False, freq: str = "YS"
```
Comment (aulemahal) [inline xclim/indices/_agro.py:1639]:
```suggestion
    tas: xarray.DataArray, positive_only: bool = False, freq: str = "YS"
```

File: xclim/indicators/atmos/_temperature.py [hunk 3]
```diff
@@ -1518,4 +1518,5 @@ def cfcheck(self, tas, snd=None):
     long_name="Chill units after the Utah Model",
     allowed_periods=["Y"],
     compute=indices.chill_units,
+    parameters=dict(positive_only={"default": False, "kind": InputKind.BOOL}),
```
Comment (aulemahal) [inline xclim/indicators/atmos/_temperature.py:1521]:
```suggestion
```
I think this line is not needed if the proper annotation is added in the function.

**Review by aulemahal — CHANGES_REQUESTED** [review]
Easy addition! A few suggestions to simplify and preserve dask's performance in the default case.

File: CHANGELOG.rst [hunk 4]
```diff
@@ -44,6 +44,10 @@ Bug fixes
 * Reorganised how ``Indicator`` subclasses can added arguments to the call signature. Injecting such arguments now works. For xclim's subclasses, this bug only affected the ``indexer`` argument of indicators subclassing ``xc.core.indicator.IndexingIndicator``. (:pull:`1981`).
 * All-nan slices are now treated correctly in method `ExtremeValues`. (:issue:`1982`, :pull:`1983`).
 
+New features and enhancements
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+* ``chill_unit`` now accepts a new argument `positive_only` to compute the daily positive chill units. (:pull:`2003`).
+
```
Comment (Zeitsperre) [inline CHANGELOG.rst:50]: This should be moved up to the latest version entry. Name should also be listed in authorship!

File: xclim/indices/_agro.py [hunk 5]
```diff
@@ -1635,17 +1635,21 @@ def chill_portions(
...
     The Utah model assigns a weight to each hour depending on the temperature recognising that high temperatures can actual decrease,
-    the potential for bud breaking.
+    the potential for bud breaking. Providing `positive_only=True` will ignore days with negative chill units.
 
     Parameters
     ----------
     tas : xr.DataArray
         Hourly temperature.
+    positive_only : bool
+        If True, only positive daily chill units are aggregated.
```
Comment (Zeitsperre) [inline xclim/indices/_agro.py:1652]:
```suggestion
        If `True`, only positive daily chill units are aggregated.
```

**Review by Zeitsperre — CHANGES_REQUESTED** [review]
Looking good. Just a few small fixes.

File: xclim/indicators/atmos/_temperature.py (hunk 3 above)
Comment (saschahofmann) [reply xclim/indicators/atmos/_temperature.py]: Makes total sense awesome 👍

**Comment by saschahofmann** [comment]
Should all be resolved

File: CHANGELOG.rst [hunk 6]
```diff
@@ -44,6 +48,7 @@ Bug fixes
 * Reorganised how ``Indicator`` subclasses can added arguments to the call signature. Injecting such arguments now works. For xclim's subclasses, this bug only affected the ``indexer`` argument of indicators subclassing ``xc.core.indicator.IndexingIndicator``. (:pull:`1981`).
 * All-nan slices are now treated correctly in method `ExtremeValues`. (:issue:`1982`, :pull:`1983`).
 
+
```
Comment (Zeitsperre) [inline CHANGELOG.rst:51]:
```suggestion
```

**Review by Zeitsperre — APPROVED** [review]
Looks good! Thanks again!

**Review by aulemahal — APPROVED** [review]
Dang sorry I forgot this. Looks all good!

**Comment by Zeitsperre** [comment]
@saschahofmann Thanks again for your help! Very much appreciate it!

**Pull request merged**